### PR TITLE
add the first test

### PR DESCRIPTION
**Pull request by lkzhao**
* add first test `FlexLayoutSpec`
* setup codecov
* add flexBasis support
* fix a bug where flexlayout doesnt shrink correctly.

### CODE REVIEW

File: Sources/Layout/CollectionLayout.swift [hunk 1]
```diff
@@ -25,7 +25,7 @@ open class CollectionLayout<Data> {
   }
   
   private var _contentSize: CGSize = .zero
-  internal func _layout(collectionSize: CGSize, dataProvider: CollectionDataProvider<Data>, sizeProvider: @escaping CollectionSizeProvider<Data>) {
+  public func _layout(collectionSize: CGSize, dataProvider: CollectionDataProvider<Data>, sizeProvider: @escaping CollectionSizeProvider<Data>) {
```
Comment (jindulys) [inline Sources/Layout/CollectionLayout.swift]: If this a public method, consider rename this method to get rid of prefixed _?

**Comment by lkzhao** [comment]
We need to rethink how we implement insets. The current way of implementing it within the superclass CollectionLayout is quite messy. The _layout method actually calls into another ‘layout’

**Comment by lkzhao** [comment]
@jindulys Test didn't pass. lets revert and wait til I fix the test.

**Comment by lkzhao** [comment]
Nevermind. will create a new PR to fix it.

**Comment by jindulys** [comment]
oops!

**Comment by lkzhao** [comment]
@jindulys It happens sometimes, no problem. I added some PR protection on github, so that next time it will block it from happening. Only PR that pass review and test can be merged into master.

**Pull request merged**